src/common/lib/ws2812/ws2812.c

```
@@ -225,81 +237,81 @@ static uint8_t g_led_brightness = CONVERTER_LEDS_BRIGHTNESS;
-    // 
-    // Performance notes:
-    // - Multiplication is fast on Cortex-M0+ (1-2 cycles for 8-bit×8-bit)
-    // - Division by 255 may be optimized by compiler to multiply + shift
-    // - Alternative: (x * multiplier * 257) >> 16 (faster but less accurate)
-    // - Current approach prioritizes accuracy over marginal speed gain
-    r = (uint8_t)((r * multiplier) / 255);
-    g = (uint8_t)((g * multiplier) / 255);
-    b = (uint8_t)((b * multiplier) / 255);
-  }
+    if (g_led_brightness <= 10) {
+        const uint8_t multiplier = BRIGHTNESS_LUT[g_led_brightness];
```

Clang (14.0.6)

[warning] variable 'multiplier' is not initialized

(cppcoreguidelines-init-variables)